src/Stratis.SmartContracts.Core/Exceptions/InsufficientGasException.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Stratis.SmartContracts.Core.Exceptions
+{
+    /// <summary>
+    /// Thrown when the remaining gas left for contract execution is less than the amount selected to allocate to an internal call. 
+    /// <para>
+    /// See <see cref="IInternalTransactionExecutor"/>.
+    /// </para>
+    /// </summary>
+    public sealed class InsufficientGasException : SmartContractException
```

Comment:
@codingupastorm seeing as this is only thrown from nested gas expenditure maybe a better name would be something like `InsufficientGasForNestedMeterException` ? Or can this exception be used elsewhere?

Reply:
It will also be used for Create calls from inside contracts, so I think that the name makes enough sense for now.

Lets change it in the future if we end up with more exceptions and this one isn't reused?